ld: error: undefined symbol: pthread_create on FreeBSD

HI,
i am trying compile on FreeBSD 13.3 and  Clang 17.0.6
but I have problem with undefined symbol with HelloWorld

```
[ 68%] Linking CXX static library libJolt.a
[ 68%] Built target Jolt
[ 68%] Building CXX object CMakeFiles/HelloWorld.dir/home/maxfx/Documents/JoltPhysics/HelloWorld/HelloWorld.cpp.o
[ 68%] Building CXX object CMakeFiles/PerformanceTest.dir/home/maxfx/Documents/JoltPhysics/PerformanceTest/PerformanceTest.cpp.o
[ 68%] Building CXX object CMakeFiles/UnitTests.dir/cmake_pch.hxx.pch
[ 69%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/ArrayTest.cpp.o
[ 69%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/FPFlushDenormalsTest.cpp.o
[ 69%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/InsertionSortTest.cpp.o
[ 69%] Linking CXX executable HelloWorld
[ 70%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/JobSystemTest.cpp.o
ld: error: undefined symbol: pthread_create
>>> referenced by __threading_support:371 (/usr/include/c++/v1/__threading_support:371)
>>>               JobSystemThreadPool.cpp.o:(std::__1::__libcpp_thread_create[abi:un170006](pthread**, void* (*)(void*), void*)) in archive libJolt.a
c++: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [CMakeFiles/HelloWorld.dir/build.make:98: HelloWorld] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/HelloWorld.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 70%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/LinearCurveTest.cpp.o
[ 71%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/PreciseMathTest.cpp.o
[ 71%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/ScopeExitTest.cpp.o
[ 72%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/StringToolsTest.cpp.o
[ 72%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Core/QuickSortTest.cpp.o
[ 73%] Building CXX object CMakeFiles/UnitTests.dir/home/maxfx/Documents/JoltPhysics/UnitTests/Geometry/ClosestPointTests.cpp.o
[ 74%] Linking CXX executable PerformanceTest
ld: error: undefined symbol: pthread_create
>>> referenced by __threading_support:371 (/usr/include/c++/v1/__threading_support:371)
>>>               JobSystemThreadPool.cpp.o:(std::__1::__libcpp_thread_create[abi:un170006](pthread**, void* (*)(void*), void*)) in archive libJolt.a
c++: error: linker command failed with exit code 1 (use -v to see invocation)
gmake[2]: *** [CMakeFiles/PerformanceTest.dir/build.make:99: PerformanceTest] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:167: CMakeFiles/PerformanceTest.dir/all] Error 2

```

--- a/Docs/ReleaseNotes.md
+++ b/Docs/ReleaseNotes.md
@@ -19,6 +19,7 @@ For breaking API changes see [this document](https://github.com/jrouwe/JoltPhysi
 * Fixed an issue where the bounding volume of a HeightFieldShape was not properly adjusted when calling SetHeights leading to missed collisions.
 * Workaround for CMake error 'CMake Error: No output files for WriteBuild!' when using the 'Ninja Multi-Config' generator.
 * When a height field was created where SampleCount / BlockSize is not a power of 2 and a soft body touched the right or bottom border of the height field, the application would crash.
+* Fixed a link error 'ld: error: undefined symbol: pthread_create' on FreeBSD
 
 ## v5.1.0
 

--- a/Jolt/Jolt.cmake
+++ b/Jolt/Jolt.cmake
@@ -686,6 +686,7 @@ endif()
 # On Unix flavors we need the pthread library
 if (NOT ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows") AND NOT EMSCRIPTEN)
 	target_compile_options(Jolt PUBLIC -pthread)
+	target_link_options(Jolt PUBLIC -pthread)
 endif()
 
 if (EMSCRIPTEN)

